Codecademy/gamut
# feat(Disclosure): creating an Disclosure component

### Overview
Adding a Disclosure component to replace the current AccordionDeprecated component. 

### PR Checklist

- [ ] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=30186-40433&m=dev
- [ ] Related to JIRA ticket: [GM-562](https://codecademy.atlassian.net/browse/GM-562)

#### Testing Instructions

1. Click on the preview link
2. Go to the `path=/docs/molecules-disclosure--disclosure`
3. Read through the documentation
4. Change around the props in the Code Playground example
(Mono) 
5. Go to the Mono Preview and visit `admin/portal` 
6. Check that the "Business Admin Users" section renders a Disclosure (aka Accordion) and loads the users upon expanding (and collapsing)
(Monolith)
7. Nothing really, check that the Preview works?
8. ...
9. Profit!

[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37851)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Mono PR       | [Mono PR](https://github.com/codecademy-engineering/mono/pull/6822)  | [Portal Env](http://www.google.fr/ 'Named link title')   |





[GM-562]: https://codecademy.atlassian.net/browse/GM-562?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
